datatypes/src/raster/geo_transform.rs

```diff
@@ -175,6 +154,61 @@ impl GeoTransform {
         }
     }
 
+    /// compute the index of the upper left pixel that is contained in the `partition`
+    pub fn upper_left_pixel_idx(&self, partition: &SpatialPartition2D) -> GridIdx2D {
+        // choose the epsilon relative to the pixel size
+        const EPSILON: f64 = 0.000_001;
```

Comment:
What is this and does this work for all conditions?
Maybe we need a test for that for different resolutions.

Reply:
i use it to move the Coordinate EPSILON * pixel_size to the reight/bottom. This solves the effect that a coordinate might be a µ pixel_size smaller that the border of the nearest pixel. We do the same thing for the lower right pixel of the box.
It should work in all conditions if the assumption that a coordinate that is a µ smaller then the border of a pixel should belong to that pixel. If that assumption does not hold then the calculation for the lower right is also wrong.